### PR TITLE
Visual for tower

### LINKED ISSUE #973: Tower visual
Visual that tells you how much energy the tower has.

### PR DESCRIPTION
**Pull request by Perrytheplatypus43**
visual for tower that displays tower contents

Resolves #973

### CODE REVIEW

**Comment by Perrytheplatypus43** [comment]
I am still developing this feature so do not merge

File: visuals.js [hunk 1]
```diff
@@ -356,6 +359,15 @@ module.exports = class Visuals {
 		}
 	}
 	
+	static tower(room) {
+	    _.forEach(room.structures.towers, tower => {
```
Comment (Spedwards) [inline visuals.js]: Move the loop to the `run` method, see spawn and sources for an example.

File: visuals.js [hunk 2]
```diff
@@ -356,6 +359,15 @@ module.exports = class Visuals {
...
+			const vis = new RoomVisual(room.name);
+			const x = 14.8;
+			let y = 3.6;
```
Comment (Spedwards) [inline visuals.js]: `const x = tower.pos.x + 1` and `const y = spawn.pos.y - 0.5`.

File: visuals.js [hunk 3]
```diff
@@ -356,6 +359,15 @@ module.exports = class Visuals {
...
+			vis.text(`Tower energy: ${tower.energy}`, x, y);
```
Comment (Spedwards) [inline visuals.js]: Set font to 0.4 and align left.

File: visuals.js (hunk 1 above)
Comment (Perrytheplatypus43) [inline visuals.js]: I moved the loop to the run method

**Comment by Perrytheplatypus43** [comment]
Did I resolve all the issues and requests? @Spedwards

**Comment by Spedwards** [comment]
@Perrytheplatypus43 You need to merge in dev to resolve the conflicts, and edit the `highlightWeakest` method.

**Comment by Spedwards** [comment]
Merge failed. You have two of every method.

**Comment by Perrytheplatypus43** [comment]
woodi

**Pull request closed without merging**